test/ir_Sanyo_test.cpp

```diff
@@ -257,3 +261,267 @@ TEST(TestDecodeSanyoLC7461, FailToDecodeNonSanyoLC7461Example) {
+  EXPECT_EQ(kSanyoAcTempMin, ac.getTemp());
+
+  ac.setTemp(kSanyoAcTempMax);
+  EXPECT_EQ(kSanyoAcTempMax, ac.getTemp());
+
+  ac.setTemp(kSanyoAcTempMin - 1);
+  EXPECT_EQ(kSanyoAcTempMin, ac.getTemp());
+
+  ac.setTemp(kSanyoAcTempMax + 1);
+  EXPECT_EQ(kSanyoAcTempMax, ac.getTemp());
+
+  ac.setTemp(17);
+  EXPECT_EQ(17, ac.getTemp());
+
+  ac.setTemp(21);
+  EXPECT_EQ(21, ac.getTemp());
+
+  ac.setTemp(25);
+  EXPECT_EQ(25, ac.getTemp());
+
+  ac.setTemp(30);
+  EXPECT_EQ(30, ac.getTemp());
+}
+
+TEST(TestSanyoAcClass, OperatingMode) {
+  IRSanyoAc ac(kGpioUnused);
+  ac.begin();
+
+  ac.setMode(kSanyoAcAuto);
+  EXPECT_EQ(kSanyoAcAuto, ac.getMode());
+
+  ac.setMode(kSanyoAcCool);
+  EXPECT_EQ(kSanyoAcCool, ac.getMode());
+
+  ac.setMode(kSanyoAcHeat);
+  EXPECT_EQ(kSanyoAcHeat, ac.getMode());
+
+  ac.setMode(kSanyoAcDry);
+  EXPECT_EQ(kSanyoAcDry, ac.getMode());
+
+  ac.setMode(kSanyoAcAuto + 1);
+  EXPECT_EQ(kSanyoAcAuto, ac.getMode());
+
+  ac.setMode(0);
+  EXPECT_EQ(kSanyoAcAuto, ac.getMode());
+
+  ac.setMode(255);
+  EXPECT_EQ(kSanyoAcAuto, ac.getMode());
+}
+
+TEST(TestSanyoAcClass, FanSpeed) {
+  IRSanyoAc ac(kGpioUnused);
+  ac.begin();
+
+  ac.setFan(kSanyoAcFanAuto);
+  EXPECT_EQ(kSanyoAcFanAuto, ac.getFan());
+
+  ac.setFan(kSanyoAcFanHigh);
+  EXPECT_EQ(kSanyoAcFanHigh, ac.getFan());
+
+  ac.setFan(kSanyoAcFanLow);
+  EXPECT_EQ(kSanyoAcFanLow, ac.getFan());
+
+  ac.setFan(kSanyoAcFanMedium);
+  EXPECT_EQ(kSanyoAcFanMedium, ac.getFan());
+}
+
+TEST(TestSanyoAcClass, Sleep) {
+  IRSanyoAc ac(kGpioUnused);
+  ac.begin();
+
+  ac.setSleep(true);
+  EXPECT_TRUE(ac.getSleep());
+  ac.setSleep(false);
+  EXPECT_FALSE(ac.getSleep());
+  ac.setSleep(true);
+  EXPECT_TRUE(ac.getSleep());
+}
+
+TEST(TestSanyoAcClass, SwingV) {
+  IRSanyoAc ac(kGpioUnused);
+  ac.begin();
+
+  ac.setSwingV(kSanyoAcSwingVAuto);
+  EXPECT_EQ(kSanyoAcSwingVAuto, ac.getSwingV());
+
+  ac.setSwingV(kSanyoAcSwingVHigh);
+  EXPECT_EQ(kSanyoAcSwingVHigh, ac.getSwingV());
+
+  ac.setSwingV(kSanyoAcSwingVLow);
+  EXPECT_EQ(kSanyoAcSwingVLow, ac.getSwingV());
+
+  ac.setSwingV(kSanyoAcSwingVUpperMiddle);
+  EXPECT_EQ(kSanyoAcSwingVUpperMiddle, ac.getSwingV());
+
+  ac.setSwingV(0);
+  EXPECT_EQ(kSanyoAcSwingVAuto, ac.getSwingV());
+  ac.setSwingV(255);
+  EXPECT_EQ(kSanyoAcSwingVAuto, ac.getSwingV());
+}
+
+TEST(TestSanyoAcClass, Timers) {
+  IRSanyoAc ac(kGpioUnused);
+  ac.begin();
+
+  ac.setOffTimer(0);
+  EXPECT_EQ(0, ac.getOffTimer());
+  ac.setOffTimer(59);
+  EXPECT_EQ(0, ac.getOffTimer());
+  ac.setOffTimer(60);
+  EXPECT_EQ(60, ac.getOffTimer());
+  ac.setOffTimer(61);
+  EXPECT_EQ(60, ac.getOffTimer());
+  ac.setOffTimer(15 * 60 + 59);
+  EXPECT_EQ(15 * 60, ac.getOffTimer());
+  ac.setOffTimer(16 * 60);
+  EXPECT_EQ(15 * 60, ac.getOffTimer());
+
+  const uint8_t offTimer2Hr[kSanyoAcStateLength] = {
+      0x6A, 0x6D, 0x4F, 0x02, 0x14, 0x85, 0x00, 0x00, 0x4A};
+  ac.setRaw(offTimer2Hr);
+  EXPECT_EQ(2 * 60, ac.getOffTimer());
```

Comment:
No `resultAcToString` tests of off timer data

Reply:
Good catch. Yes, I was lazy. Fixed/done.